### ISSUE TITLE
Having trouble with login via command line parameters

### ISSUE THREAD

**Issue by TrevorPT** [mseminatore/TeslaJS#60]
Hi, I noticed you recently implemented the .token file for stateful logins, but I'm having trouble using the command-line parameters -u and -p. I'm trying this because simpleStreaming.js doesn't seem to work with the .token file (I'll open a separate issue).  If I specify -u and -p for one of the examples, like odometer.js, I get the following output:

```javascript
F:\projects\teslajs\node_modules\teslajs\Examples\sampleFramework.js:34
        if (result.error) {
                  ^

TypeError: Cannot read property 'error' of null
    at SampleFramework.login_cb (F:\projects\teslajs\node_modules\teslajs\Exampl
es\sampleFramework.js:34:19)
    at Request._callback (F:\projects\teslajs\node_modules\teslajs\TeslaJS.js:18
9:9)
    at Request.self.callback (F:\projects\teslajs\node_modules\teslajs\node_modu
les\request\request.js:186:22)
    at emitTwo (events.js:106:13)
    at Request.emit (events.js:192:7)
    at Request.<anonymous> (F:\projects\teslajs\node_modules\teslajs\node_module
s\request\request.js:1081:10)
    at emitOne (events.js:96:13)
    at Request.emit (events.js:189:7)
    at IncomingMessage.<anonymous> (F:\projects\teslajs\node_modules\teslajs\nod
e_modules\request\request.js:1001:12)
    at Object.onceWrapper (events.js:291:19)
```

It looks like `login_cb` in `function SampleFramework.` is only expecting `(result)` as a parameter.

`function login` however is invoking the callback as: 
```javascript
callback(error, { error: error, response: response, body: body, authToken: authToken });
```

I'm thinking the line in `sampleFramework.js` should be
```javascript
this.login_cb = function (error, result) {
```

Is this work in progress or do I have something wrong?

I installed the latest from npm but according to package.json I have version 2.1.7.  Thanks.

**Comment by TrevorPT** [comment]
Additional: I deleted teslajs and reinstalled from npm, and now I have verison 2.1.10.  I thought `npm update` would have updated it to the latest version?

**Comment by mseminatore** [comment]
@TrevorPT `npm update` will update your dependencies to the latest version following the `semver` rules in your project's package.json file.  Have you created your own project using `npm init` followed by `npm install teslajs --save`?

**Comment by mseminatore** [comment]
@TrevorPT I'll take a look at the streaming sample.  It is possible that something that changed with a prior refactoring broke the sample and went unnoticed.

**Comment by mseminatore** [comment]
@TrevorPT Thanks for reporting this, I've found an issue with the simpleStreaming sample that was a left-over from migrating to the new sample framework.  I am working on a fix right now and should have this checked in shortly.

**Comment by mseminatore** [comment]
@TrevorPT OK, fix checked in **2.1.11**.  The `.token` file to cache the OAuth token has been around from the beginning.  The streaming API works differently than other API's.  It takes a token that is returned from the call to enumerate vehicle information.  This token is used as the "password" and is paired with the username.

**Comment by TrevorPT** [comment]
@mseminatore - thanks! I'll give it another try with the update and continue with the streaming thread in #61.  Regarding my question about using npm to update, I'm embarrassed to say I should've known as (believe it or not) I've used npm before! I'm still just playing with the examples but I keep thinking of TeslaJS as the project, not as a module, so please excuse the thoughtless questions!
